explorer/src/metaplex/Art/Art.tsx

```diff
@@ -0,0 +1,246 @@
+import { pubkeyToString } from "../utils";
+import { useCachedImage, useExtendedArt } from "./useArt";
+import { Stream, StreamPlayerApi } from "@cloudflare/stream-react";
+import { PublicKey } from "@solana/web3.js";
+import { getLast } from "../utils";
+import { Metadata } from "metaplex/classes";
+
+const CachedImageContent = ({
+  uri,
+  className,
+  preview,
+  style,
+}: {
+  uri?: string;
+  className?: string;
+  preview?: boolean;
+  style?: React.CSSProperties;
+}) => {
+  const { cachedBlob } = useCachedImage(uri || "");
+
+  return (
+    <Image
+      src={cachedBlob}
+      preview={preview}
+      wrapperClassName={className}
+      loading="lazy"
+      wrapperStyle={{ ...style }}
+      style={{ width: 150, borderRadius: 12 }}
+      height={"auto"}
+    />
+  );
+};
+
+const VideoArtContent = ({
+  className,
+  style,
+  files,
+  uri,
+  animationURL,
+  active,
+}: {
+  className?: string;
+  style?: React.CSSProperties;
+  files?: (MetadataFile | string)[];
+  uri?: string;
+  animationURL?: string;
+  active?: boolean;
+}) => {
+  const [playerApi, setPlayerApi] = useState<StreamPlayerApi>();
+
+  const playerRef = useCallback(
+    (ref) => {
+      setPlayerApi(ref);
+    },
+    [setPlayerApi]
+  );
+
+  useEffect(() => {
+    if (playerApi) {
+      playerApi.currentTime = 0;
+    }
+  }, [active, playerApi]);
+
+  const likelyVideo = (files || []).filter((f, index, arr) => {
+    if (typeof f !== "string") {
+      return false;
+    }
+
+    // TODO: filter by fileType
+    return arr.length >= 2 ? index === 1 : index === 0;
+  })?.[0] as string;
+
+  const content =
+    likelyVideo &&
+    likelyVideo.startsWith("https://watch.videodelivery.net/") ? (
+      <div className={`${className} square`}>
+        <Stream
+          streamRef={(e: any) => playerRef(e)}
+          src={likelyVideo.replace("https://watch.videodelivery.net/", "")}
+          loop={true}
+          height={150}
+          width={150}
+          controls={false}
+          style={{ borderRadius: 12 }}
+          videoDimensions={{
+            videoHeight: 150,
+            videoWidth: 150,
+          }}
+          autoplay={true}
+          muted={true}
+        />
+      </div>
+    ) : (
+      <video
+        className={className}
+        playsInline={true}
+        autoPlay={true}
+        muted={true}
+        controls={true}
+        controlsList="nodownload"
+        style={{ borderRadius: 12, ...style }}
+        loop={true}
+        poster={uri}
+      >
+        {likelyVideo && (
+          <source src={likelyVideo} type="video/mp4" style={style} />
+        )}
+        {animationURL && (
+          <source src={animationURL} type="video/mp4" style={style} />
+        )}
+        {files
+          ?.filter((f) => typeof f !== "string")
+          .map((f: any) => (
+            <source src={f.uri} type={f.type} style={style} />
+          ))}
+      </video>
+    );
+
+  return content;
+};
+
+const HTMLContent = ({
+  uri,
```

Comment:
```suggestion
```

Reply:
Wow that did not format it how I wanted it too. Going to try that again. Apologies